### PR TITLE
Add FashionMNIST dataset

### PR DESCRIPTION
**Pull request by gchhablani**
This PR adds [FashionMNIST](https://github.com/zalandoresearch/fashion-mnist) dataset.

### CODE REVIEW

File: datasets/fashion_mnist/README.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+  - mit
+multilinguality: []
+size_categories:
+  - 10K<n<100K
+source_datasets:
+  - original
+task_categories:
+  - other
+task_ids:
+  - other-other-image-classification
```
Comment (lhoestq) [inline datasets/fashion_mnist/README.md:16]:
```suggestion
annotations_creators:
- expert-generated
language_creators: []
languages: []
licenses:
- mit
multilinguality: []
size_categories:
- 10K<n<100K
source_datasets:
- original
task_categories:
- other
task_ids:
- other-other-image-classification
```

File: datasets/fashion_mnist/README.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+### Data Splits
+
+The data is split into training and test set. The training set contains 60,000 images and the test set 10,000 images.
+
+## Dataset Creation
+
+### Curation Rationale
+
+The original MNIST dataset contains a lot of handwritten digits. Members of the AI/ML/Data Science community love this dataset and use it as a benchmark to validate their algorithms. In fact, MNIST is often the first dataset researchers try. "If it doesn't work on MNIST, it won't work at all", they said. "Well, if it does work on MNIST, it may still fail on others."
```
Comment (lhoestq) [inline datasets/fashion_mnist/README.md:97]: This paragraph comes from kaggle right ? Or maybe another website ?
Would be nice to make it explicit that this comes from kaggle.

File: datasets/fashion_mnist/fashion_mnist.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+            ),
+            supervised_keys=("image", "label"),
+            homepage="https://github.com/zalandoresearch/fashion-mnist",
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        urls_to_download = {key: _URL + fname for key, fname in _URLS.items()}
+        downloaded_files = dl_manager.download_and_extract(urls_to_download)
+        print(downloaded_files)
```
Comment (lhoestq) [inline datasets/fashion_mnist/fashion_mnist.py:108]:
```suggestion
```
remove the print :)

**Review by lhoestq — COMMENTED** [review]
Really nice thank you !

File: datasets/fashion_mnist/README.md (hunk 2 above)
Comment (gchhablani) [reply datasets/fashion_mnist/README.md:97]: This paragraph comes from the paper on arXiv. Infact, all information is here from the paper.

File: datasets/fashion_mnist/fashion_mnist.py (hunk 3 above)
Comment (gchhablani) [reply datasets/fashion_mnist/fashion_mnist.py:108]: Ah, sorry ;-;

File: datasets/fashion_mnist/README.md (hunk 2 above)
Comment (lhoestq) [reply datasets/fashion_mnist/README.md:98]: Oh indeed you're right, nevermind

**Comment by gchhablani** [comment]
Hi @lhoestq,

I have added the changes from the review.

File: datasets/fashion_mnist/README.md (hunk 2 above)
Comment (gchhablani) [reply datasets/fashion_mnist/README.md:98]: I added "From the arXiv paper" everywhere.

**Review by lhoestq — APPROVED** [review]
Awesome thanks !!

**Pull request merged**